[Review Only] staging:iio:ad7606: update driver: add ad7605-4 + some fixes/updates

commodo opened the pull request:
The intent is to move the driver out of staging upstream, and maybe later add some more devices from this family.
Currently, this changeset add support for AD7605-4, does some minor cleanups/docs update, and fixes the voltage scales.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

lclausen-adi commented on drivers/staging/iio/adc/ad7606.c:
```diff
@@ -26,9 +26,11 @@
 #include "ad7606.h"
 
-/* Scales are computed as 2.5/2**16 and 5/2**16 respectively */
+/* Scales are computed as 5000/2**16 and 10000/2**16 respectively,
```
Either 10000 and 20000 / 2**16 or 5000 and 10000 / 2** 16

commodo replied:
ah, right;
thanks for catching this

commodo replied:
will just leave 32768; since it's from the datasheet

lclausen-adi commented on drivers/staging/iio/adc/ad7606.h:
```diff
@@ -11,20 +11,39 @@
+ *			controls power-down mode of device
+ * @gpio_frstdata	GPIO descriptor for reading from device when data
+ *			is being read on the first channel
+ * @gpio_os		GPIO descriptor to control oversampling on the device
```
descriptors

lclausen-adi commented on drivers/staging/iio/adc/ad7606.h:
```diff
@@ -11,20 +11,39 @@
+ * @range		voltage range selection, selects which scale to apply
+ * @oversampling	oversampling selection
+ * @done		marks whether reading data is done
+ * @base_address	address from where to read data in Parallel operation
```
parallel with a lower case p

commodo commented:
Changelog v1 -> v2:
* updated doc annotations with review comments
* updated comment about scale computation
* added new patch `staging:iio:ad7606: update api calls to devm_* variants `
* updated author to myself

commodo commented:
Changelog v2 -> v3:
* fix checkpatch complaint about parameter alignment in commit `staging:iio:ad7606: update api calls to devm_*`
